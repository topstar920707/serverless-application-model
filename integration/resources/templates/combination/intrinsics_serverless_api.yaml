Parameters:
  Bucket:
    Type: String
  CodeKey:
    Type: String
  SwaggerKey:
    Type: String
  MyStageName:
    Type: String
    Default: devstage
  CacheClusterEnabled:
    Type: String
    Default: "true"

Conditions:
  TrueCondition:
    Fn::Equals:
      - true
      - true
  FalseCondition:
    Fn::Equals:
      - true
      - false

Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        GetApi:
          Type: Api
          Properties:
            Path: /pathget
            Method: get
            RestApiId:
              Ref: MyApi

        PostApi:
          Type: Api
          Properties:
            Path: /pathpost
            Method: post
            RestApiId:
              Ref: MyApi

      Tags:
        TagKey1:
          Ref: MyStageName

  MyLambdaFunctionFalseCondition:
    Type: AWS::Serverless::Function
    Condition: FalseCondition
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        GetApi:
          Type: Api
          Properties:
            Path: /pathget
            Method: get
            RestApiId:
              Ref: MyApi

        PostApi:
          Type: Api
          Properties:
            Path: /pathpost
            Method: post
            RestApiId:
              Ref: MyApi

      Tags:
        TagKey1:
          Ref: MyStageName

  MyApi:
    Type: AWS::Serverless::Api
    Condition: TrueCondition
    Properties:
      StageName:
        Ref: MyStageName
      DefinitionUri:
        Bucket:
          Ref: Bucket
        Key:
          Ref: SwaggerKey
      Variables:
        Var1:
          "Fn::Join": ["", ["a", "b"]]
        Var2:
          "Fn::Join": ["", ["1", "2"]]

  MyApiFalseCondition:
    Type: AWS::Serverless::Api
    Condition: FalseCondition
    Properties:
      StageName:
        Ref: MyStageName
      DefinitionUri:
        Bucket:
          Ref: Bucket
        Key:
          Ref: SwaggerKey
      Variables:
        Var1:
          "Fn::Join": ["", ["a", "b"]]
        Var2:
          "Fn::Join": ["", ["1", "2"]]

