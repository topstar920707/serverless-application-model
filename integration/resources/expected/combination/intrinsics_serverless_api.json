[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionPostApiPermissionStage", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyLambdaFunctionGetApiPermissionStage", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyApiDeployment", "ResourceType":"AWS::ApiGateway::Deployment" },
  { "LogicalResourceId":"MyApi", "ResourceType":"AWS::ApiGateway::RestApi" },
  { "LogicalResourceId":"MyApiStage", "ResourceType":"AWS::ApiGateway::Stage" }
]